src/main/java/fi/helsinki/cs/tmc/core/commands/DownloadModelSolution.java

```diff
@@ -41,10 +41,9 @@ public Boolean call() throws Exception {
 
         String courseName = getCourseName();
 
-        Path target =
-                Paths.get(
-                        exerciseDownloader.createCourseFolder(
-                                settings.getTmcMainDirectory(), courseName));
+        Path target
+                = exerciseDownloader.createCourseFolder(
+                        Paths.get(settings.getTmcMainDirectory()), courseName);
```

Comment:
Will this result in NPE if `settings.getTmcMainDirectory()` returns `null`? 

Checking for `null` every time `settings.getTmcMainDirectory()` is called seems cumbersome. Do we have a nice way of declaring that either the interface must always return a non-null value or otherwise doing this check in one place only?